src/Storages/MergeTree/MergeTreePartInfo.h

```diff
@@ -191,4 +191,21 @@ struct DetachedPartInfo : public MergeTreePartInfo
 
 using DetachedPartsInfo = std::vector<DetachedPartInfo>;
 
+using PartitionIds = std::vector<String>;
+
+void compactPartitionIds(PartitionIds &);
+
+inline bool containsInPartitionIdsOrEmpty(const PartitionIds & haystack, const String & needle)
+{
+    switch (haystack.size())
```

Comment:
Maybe a simple if would do here? I'm not sure that hand-optimizing the 1-item haystack case actually pays off.

Reply:
Ok, then it might be worth adding a comment explaining why we need such optimization